### PR TITLE
Implements workflow to Airflow DAG compilation

### PR DESCRIPTION
**Pull request by saurav-c**
## Describe your changes and why you are making these changes
This PR implements the Python job for compiling an Aqueduct workflow into an Airflow DAG. The Python job is responsible for generating a Python file that can be interpreted by the Airflow server.

## Related issue number (if any)
ENG 877

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

Demo: https://www.loom.com/share/8797a77a08f94ffa969ff4471ffcc8b0

### CODE REVIEW

File: src/python/aqueduct_executor/operators/airflow/dag.template [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    if spec_type == enums.JobType.FUNCTION:
+        spec = func_spec.FunctionSpec(**spec)
+        handle_function(spec, dag_run_id)
+    elif spec_type == enums.JobType.EXTRACT:
+        spec = conn_spec.ExtractSpec(**spec)
+        handle_extract(spec, dag_run_id)
+    elif spec_type == enums.JobType.LOAD:
+        spec = conn_spec.LoadSpec(**spec)
+        handle_load(spec, dag_run_id)
+    elif spec_type == enums.JobType.PARAM:
```
Comment (saurav-c) [inline src/python/aqueduct_executor/operators/airflow/dag.template:31]: TODO: Add support for system metrics

File: src/python/aqueduct_executor/operators/airflow/dag.template [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    param_execute.run(spec)
+
+
+with DAG(
+    dag_id='{{ dag_id }}',
+    default_args={
+        'retries': 0,
+    },
+    start_date=datetime(2022, 1, 1, 1),
```
Comment (saurav-c) [inline src/python/aqueduct_executor/operators/airflow/dag.template:88]: TODO: Add to Airflow cleanup task. Set proper start_date

File: src/python/aqueduct_executor/operators/airflow/execute.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    """
+    Takes a CompileAirflowSpec and generates an Airflow DAG specification Python file.
+    It returns the DAG file.
+    """
+
+    # Init Airflow tasks
+    tasks = []
+    task_to_alias = {}
+    i = 0
+    for task_id, task_spec in spec.task_specs.items():
```
Comment (saurav-c) [inline src/python/aqueduct_executor/operators/airflow/execute.py:57]: TODO: Consider topologically sorting tasks before generating aliases

File: src/python/aqueduct_executor/operators/function_executor/execute.py [hunk 4]
```diff
@@ -166,3 +171,29 @@ def run(spec: FunctionSpec) -> None:
...
+
+    extract_function.run(spec)
+
+    requirements_path = os.path.join(op_path, "requirements.txt")
+    if os.path.exists(requirements_path):
+        os.system("pip3 install -r {}".format(requirements_path))
+    
+    run(spec)
+
+    if fn_extract_path:
```
Comment (saurav-c) [inline src/python/aqueduct_executor/operators/function_executor/execute.py:196]: TODO: Figure out how to perform cleanup if there is an error before this point

File: src/golang/lib/airflow/utils.go [hunk 5]
```diff
@@ -5,11 +5,35 @@ import (
...
 }
 
 // generateTaskId generates an Airflow task ID for an operator.
-func generateTaskId(operatorName string) string {
-	return strings.ReplaceAll(operatorName, " ", "_")
+func generateTaskId(operatorName string) (string, error) {
+	return prepareId(operatorName)
+}
+
+// prepareId replaces all non-alphanumeric characters in `s` with
```
Comment (hsubbaraj-spiral) [inline src/golang/lib/airflow/utils.go:17]: This must have been a pain to do. Are there other examples of projects converting operators into airflow tasks that also do this?

**Review by hsubbaraj-spiral — APPROVED** [review]
This is great! I can't confirm the airflow spec details, but the conversion logic all looks good to me!

File: src/golang/lib/airflow/utils.go (hunk 5 above)
Comment (saurav-c) [reply src/golang/lib/airflow/utils.go:17]: Hmm I actually haven't come across other projects generating Airflow scripts

File: src/python/aqueduct_executor/operators/function_executor/execute.py (hunk 4 above)
Comment (saurav-c) [reply src/python/aqueduct_executor/operators/function_executor/execute.py:225]: https://linear.app/aqueducthq/issue/ENG-1496/clean-up-extracted-function-files-in-airflow-execution

**Pull request merged**